components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/authorization/DeviceAccessAuthorizationServiceTest.java

```diff
@@ -0,0 +1,356 @@
+    private static final String PASSWORD = "password";
+    private DeviceAccessAuthorizationServiceImpl deviceAccessAuthorizationService;
+    private List<DeviceIdentifier> deviceIds = new ArrayList<>();
+    private List<DeviceIdentifier> groupDeviceIds = new ArrayList<>();
+    private Map<String, String> defaultUserClaims;
+
+    @BeforeClass
+    public void init() throws Exception {
+        DeviceConfigurationManager.getInstance().initConfig();
+        log.info("Initializing test environment to test DeviceAccessAuthorization Class");
+        for (int i = 0; i < NO_OF_DEVICES; i++) {
+            deviceIds.add(new DeviceIdentifier(DEVICE_ID_PREFIX + i, DEVICE_TYPE));
+        }
+        List<Device> devices = TestDataHolder.generateDummyDeviceData(this.deviceIds);
+        DeviceManagementProviderService deviceMgtService = new DeviceManagementProviderServiceImpl();
+        DeviceManagementServiceComponent.notifyStartupListeners();
+        DeviceManagementDataHolder.getInstance().setDeviceManagementProvider(deviceMgtService);
+        DeviceManagementDataHolder.getInstance().setRegistryService(getRegistryService());
+        DeviceManagementDataHolder.getInstance().setGroupManagementProviderService(new
+                GroupManagementProviderServiceImpl());
+        DeviceManagementDataHolder.getInstance().setDeviceTaskManagerService(null);
+        deviceMgtService.registerDeviceType(new TestDeviceManagementService(DEVICE_TYPE,
+                MultitenantConstants.SUPER_TENANT_DOMAIN_NAME));
+        for (Device device : devices) {
+            deviceMgtService.enrollDevice(device);
+        }
+        List<Device> returnedDevices = deviceMgtService.getAllDevices(DEVICE_TYPE);
+        for (Device device : returnedDevices) {
+            if (!device.getDeviceIdentifier().startsWith(DEVICE_ID_PREFIX)) {
+                throw new Exception("Incorrect device with ID - " + device.getDeviceIdentifier() + " returned!");
+            }
+        }
+        deviceAccessAuthorizationService = Mockito.mock(DeviceAccessAuthorizationServiceImpl.class,
+                Mockito.CALLS_REAL_METHODS);
+        defaultUserClaims = buildDefaultUserClaims(FIRST_NAME, LAST_NAME, EMAIL);
+        initializeTestEnvironment();
+        //Starting tenant flow
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(MultitenantConstants.SUPER_TENANT_ID, true);
+    }
+
+    private RegistryService getRegistryService() throws RegistryException, UserStoreException {
+        RealmService realmService = new InMemoryRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(ADMIN_USER);
+        BasicDataSource dataSource = new BasicDataSource();
+        String connectionUrl = "jdbc:h2:./target/databasetest/CARBON_TEST";
+        dataSource.setUrl(connectionUrl);
+        dataSource.setDriverClassName("org.h2.Driver");
```

Comment:
Let's define a constant for org.h2.Driver ?

Reply:
+1 fixed in 9ef622da4d64b16865314ac54c0af3d1e208cbcc